src/modules/menu/components/advancedTools/ImportFile/utils.js

```diff
@@ -39,31 +41,57 @@ export function isGpx(fileContent) {
  * @param {OBject} store Vuex store
  * @param {ArrayBuffer} content Content of the file
  * @param {string} source Source of the file (either URL or file path)
+ * @param {File | null} [originalFile]
  * @returns {ExternalLayer} External layer object
  */
-export async function handleFileContent(store, content, source) {
+export async function handleFileContent(store, content, source, originalFile = null) {
     let layer = null
-    let textContent
+    let parsedContent
     let linkFiles
     if (isZipContent(content)) {
         log.debug(`File content is a zipfile, assume it is a KMZ archive`)
         const kmz = await unzipKmz(content, source)
-        textContent = kmz.kml
+        parsedContent = kmz.kml
         linkFiles = kmz.files
+    } else if (source.endsWith('.tif')) {
+        log.debug(`File content might be a COGTIFF, attempting a parse as such`)
```

Comment:
I'm not sure it is a good idea to test for the extension, we don't do this for GPX and KML.
Maybe we should rethink this logic here and use the same as for zip file by checking the magic number header of tiff file (`49 49 2A 00` or `4D 4D 00 2A` depending on endianness)

Reply:
What about for people using `.tiff` extension or saving the file without extension on a server ?

Reply:
how do you find out which byes comes first in file? is there a website for that?

Reply:
I used for zip file detection the same approach using the header magic number without being concerned about endianness which might be a mistake, don't know